admin/subtotal_setup.php

```diff
@@ -179,7 +179,12 @@
 // Activer la gestion des blocs "Non Compris" pour exclusion du total
 $formSetup->newItem('ManageNonCompris')->setAsTitle();
 
-$formSetup->newItem('SUBTOTAL_MANAGE_COMPRIS_NONCOMPRIS')->setAsYesNo();
+$itemNC = $formSetup->newItem('SUBTOTAL_MANAGE_COMPRIS_NONCOMPRIS')->setAsSelect(array(0 => $langs->transnoentities('No'), 1 => $langs->transnoentities('Yes')));
+$itemNC->setSaveCallBack(function ($itemNC){
+	if((int) $itemNC->fieldValue > 0) {
+		_createExtraComprisNonCompris();
+	}
+});
```

Comment:
Eu pourquoi tu fait pas ta sauvegarde dans justement le save callback ?
```suggestion
$itemNC->setSaveCallBack(function ($itemNC){
         $result = dolibarr_set_const($itemNC->db, $itemNC->confKey, $itemNC->fieldValue, 'chaine', 0, '', $itemNC->entity);

	if((int) $itemNC->fieldValue > 0) {
		_createExtraComprisNonCompris();
	}
	
	return $result;
});
```

Reply:
a vérifier j'ai mis la suggestion a titre d'exemple